Libraries/Utilities/DictionaryExtensions.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+
+namespace System.Collections.Generic
+{
+    public static class DictionaryExtensions
+    {
+        /// <summary>
+        /// Safely returns the value by key if it's in the dictionary. Otherwise, it'll return <see langword="null" />.
+        /// </summary>
+        /// <param name="key">Key in the dictionary.</param>
+        /// <typeparam name="T">Type of the values in the dictionary.</typeparam>
+        /// <returns>Value identified by the key if it's in the dictionary.</returns>
+        public static T GetMaybe<T>(this IDictionary<string, T> dictionary, string key) =>

```

Comment:
Miért kell ez külön a generic változat mellett?

Reply:
Igazából ezt @DAud-IcI írta, én csak átmásoltam, és ez a kérdés bennem is felmerült. Tuti van valami oka.

Reply:
Genericnél ott a `notnull` constaint, ez az egyetlen különbség, de valójában a stringes változatnál sem lehet `null` a key (mert exception lesz), szóval a generic igazából jobb is a compile-time checkkel. Kitöröltem a stringeset és minden okénak tűnik.

Reply:
Emiatt most failel egy rahedli teszt, mert a key null. Szerintem itt valami hasonló járhatott Dávid fejében még ezzel a GetMaybe-vel (legalább a stringes overloadnál), hogy lehet null a string és akkor nem lesz katasztrófa. Lehetne itt a GetMaybe-hez egy olyan, hogy levesszük a notnull constrain-t, és ha null (vagy default) a key, akkor nullt adunk vissza. Kicsit meredek lenne az összes GetMaybe hívásnál nullchecket csinálni, és igazából mondhatjuk azt, hogy a GetMaybe-nek ez a másik specialitása, hogy elfogad null keyt exception dobása nélkül.

Reply:
A null visszaadása null key esetén OK.